Jellyfin.Server/Program.cs

```diff
@@ -0,0 +1,307 @@
+                OperatingSystem = getOperatingSystem()
+            };
+
+        private static MediaBrowser.Model.System.OperatingSystem getOperatingSystem() {
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.MacOSX:
+                    return MediaBrowser.Model.System.OperatingSystem.OSX;
+                case PlatformID.Win32NT:
+                    return MediaBrowser.Model.System.OperatingSystem.Windows;
+                case PlatformID.Unix:
+                default:
+                {
+                    string osDescription = RuntimeInformation.OSDescription;
+                    if (osDescription.Contains("linux", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return MediaBrowser.Model.System.OperatingSystem.Linux;
+                    }
+                    else if (osDescription.Contains("darwin", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return MediaBrowser.Model.System.OperatingSystem.OSX;
+                    }
+                    else if (osDescription.Contains("bsd", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return MediaBrowser.Model.System.OperatingSystem.BSD;
+                    }
+                    throw new Exception($"Can't resolve OS with description: '{osDescription}'");
+                }
+            }
+        }
+
+        public static  void Shutdown()
+        {
+            ApplicationTaskCompletionSource.SetResult(true);
+        }
+
+        public static void Restart()
+        {
+            _restartOnShutdown = true;
+
+            Shutdown();
+        }
+
+        private static void StartNewInstance(StartupOptions startupOptions)
+        {
+            _logger.LogInformation("Starting new instance");
+
+            string module = startupOptions.GetOption("-restartpath");
+
+            if (string.IsNullOrWhiteSpace(module))
+            {
+                module = Environment.GetCommandLineArgs().First();
+            }
+
+            string commandLineArgsString;
+
+            if (startupOptions.ContainsOption("-restartargs"))
+            {
+                commandLineArgsString = startupOptions.GetOption("-restartargs") ?? string.Empty;
+            }
+            else
+            {
+                commandLineArgsString = string .Join(" ", 
+                    Environment.GetCommandLineArgs()
+                        .Skip(1)
+                        .Select(NormalizeCommandLineArgument)
+                    );
+            }
+
+            _logger.LogInformation("Executable: {0}", module);
+            _logger.LogInformation("Arguments: {0}", commandLineArgsString);
+
+            Process.Start(module, commandLineArgsString);
+        }
+
+        private static string NormalizeCommandLineArgument(string arg)
+        {
+            if (!arg.Contains(" ", StringComparison.OrdinalIgnoreCase))

```

Comment:
Isn't there some builtin dotnet logic for arg escaping? The one below would fail for args containing double quotes at least.

Reply:
Actually I would say that this is the only real blocker, others could be resolved later if you want.

Reply:
IDK, This hasn't changed, still the same logic as before.

Reply:
Good ol' bug then... Still worth fixing I daresay

Reply:
@Bond-009 were you able to look into this?

Reply:
@JustAMan I looked around a little, and it seems properly escaping args isn’t that easy.

Reply:
Let's deal with it when it is reported as an issue then.